salt/modules/file.py

```diff
@@ -3213,6 +3213,62 @@ def touch(name, atime=None, mtime=None):
     return os.path.exists(name)
 
 
+def tail(path, lines):
+    '''
+    .. versionadded:: Neon
+
+    Read the last n lines from a file
+
+    path
+        path to file
+
+    lines
+        number of lines to read
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' file.tail /path/to/file 10
+    '''
+    path = os.path.expanduser(path)
+    lines_found = []
+
+    if not os.path.isfile(path):
+        raise SaltInvocationError('File not found: {0}'.format(path))
+
+    if not __utils__['files.is_text'](path):
+        raise SaltInvocationError(
+            'Cannot tail a binary file: {0}'.format(path))
+
+    try:
+        lines = int(lines)
+    except ValueError:
+        log.error('file.tail: \'lines\' value must be an integer')
+        lines = 10
```

Comment:
I don't understand why you set `lines = 10` here. I would suggest raising an exception instead (in which case you can remove `log.error`).

Reply:
It's a safe value to default to in case they formed their command wrong.

I'm not attached to doing this way, I can change it if it's a problem.

Reply:
It is not a good value on Windows because `next_char = tail_fh.read(1)` will always return 1 character and you are comparing a 2-character string, which will never test positive.

Reply:
Sorry, my previous comment belonged to the other issue I opened. As for this case, it doesn't make sense to use a default in the case of invalid input. I think an exception is the best answer for invalid input. Defaults should be used for cases where there is no input, which isn't the case here.